Webapp/sources/blueprints/reagents/routes.py

```diff
@@ -31,62 +31,34 @@ def reagents() -> Response:
-        ):  # check the novel compound db for the cas number
-            found_reagent = (
-                db.session.query(models.NovelCompound)
-                .filter(models.NovelCompound.cas == reagent)
-                .join(models.WorkBook)
-                .filter(models.WorkBook.id == workbook.id)
-                .first()
-            )
+    cas_match = re.findall(cas_regex, reagent)
+    if cas_match:
+        cas_number = cas_match[0]
+        found_reagent = services.all_compounds.from_cas(cas_number, workbook)
         if found_reagent is None:
             # if it's a cas but not in database then return for adding novel compound redirect
             return jsonify({"cas_not_found": True, "reagent": reagent})
     else:  # non-cas entry. We search reagent common names
-        found_reagent = (
-            db.session.query(models.Compound)
-            .filter(func.lower(models.Compound.name) == reagent.lower())
-            .first()
+        found_reagent = services.all_compounds.from_name(reagent, workbook)
+        if isinstance(found_reagent, models.NovelCompound):
+            novel_compound = True
+
+    # If still no match find a list of partial matches and send to the frontend
+    if found_reagent is None:
+        # If working in a workbook, get its novel compounds and recently used compounds
+        reagent_names = services.all_compounds.populate_reagent_dropdown(
+            reagent, workbook
+        )
+        return jsonify(
+            {
+                "identifiers": reagent_names,
+                "reagent_not_found": True,
+                "reagent": reagent,
+            }
         )
-        if found_reagent is None and workbook:  # check novel compound db
-            found_reagent = (
-                db.session.query(models.NovelCompound)
-                .filter(func.lower(models.NovelCompound.name) == reagent.lower())
-                .join(models.WorkBook)
-                .filter(models.WorkBook.id == workbook.id)
-                .first()
-            )
-            if found_reagent:
-                novel_compound = True
-        if (
-            found_reagent is None
-        ):  # If still no match find a list of partial matches and send to the frontend
-            reagent_list = (
-                db.session.query(models.Compound)
-                .filter(func.lower(models.Compound.name).startswith(reagent.lower()))
-                .order_by(models.Compound.name.asc())
-                .limit(100)
-                .all()
-            )
-            reagent_names = [x.name for x in reagent_list]
-            return jsonify(
-                {
-                    "identifiers": reagent_names,
-                    "reagent_not_found": True,
-                    "reagent": reagent,
-                }
-            )
     if found_reagent is not None:
+        if workbook and not novel_compound:
+            services.workbook.add_compound_to_recent_list(workbook, found_reagent)
```

Comment:
are novel_compounds added at a different point?

Reply:
Novel compounds for a workbook are added to the top of the list at the moment then recently used reagents after that.

Reply:
ahh ok, I assumed that would be the case but thought I'd point it out in case